### PR TITLE
[IGNORE] Query schemas: update load logic + validation logic + unit tests according to query datamodel breaking change

### PR DESCRIPTION
**Pull request by AntoineThebaud**
This query datamodel breaking change actually involved to rework quite some logic on the backend side regarding schemas load and schemas validation. Actually it simplified a lot of things, that we had to originally put in place to cope with the former "open" datamodel for queries ('could be in a single object, an array or even a map, located anywhere at any depth within the panel plugin spec). Now that the backend code is "aware" of where the queries are located, the logic for those could be realigned on what we have for all the other types of plugins (i.e panels, variables & datasources) for which this was already the case.

_NB: cue doc to be updated accordingly in another PR - dont want to create merge conflict because of this for now._

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).
- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.

### CODE REVIEW

**Comment by AntoineThebaud** [comment]
@eunicorn I opened this PR in top of your branch
@Nexucis I was assuming eunice's [BREAKINGCHANGE] tag on her "main" branch for this whole change is enough, hence why I put IGNORE here, but let me know if this is wrong

**Comment by Nexucis** [comment]
> @Nexucis I was assuming eunice's [BREAKINGCHANGE] tag on her "main" branch for this change is enough, hence why I put IGNORE here, but let me know if this is wrong

That's a good questions, I'm even wondering if we should squash her PR or merge it actually to preserve the different commit

**Review by Nexucis — APPROVED** [review]

File: internal/api/shared/migrate/testdata/old_grafana_panels_perses_dashboard.json [hunk 1]
```diff
@@ -47,7 +47,8 @@
             "spec": {
               "text": "<p>This dashboard provides information on OTF Front-End queues.<br/>Minimum OTF FE version required: 18.0.0.68 / 19.0.0.29<br/>\nMore information on FE metrics used here: <a href=\"https://rndwww.nce.amadeus.net/confluence/display/OTFTeam/OTF+Monitoring+-+FE+metrics\">Confluence link</a>\n</p>"
             }
-          }
+          },
+          "queries": []
```
Comment (Nexucis) [inline internal/api/shared/migrate/testdata/old_grafana_panels_perses_dashboard.json:51]: mmm this should not appear if it is empty. But I guess it's coming from the other branch. I will look there

Comment (AntoineThebaud) [reply internal/api/shared/migrate/testdata/old_grafana_panels_perses_dashboard.json:51]: Hmm if you really want it to not generate an empty array when there's no query, there's most probably a new check to add in the cue logic here https://github.com/perses/perses/pull/1072/files#diff-d4979002465bf1e3768d854a8ba3c0caa4802753bf17e356b3c3a6015b3460ea

**Comment by eunicorn** [comment]
> Actually it simplified a lot of things, that we had to originally put in place to cope with the former "open" datamodel for queries ('could be in a single object, an array or even a map, located anywhere at any depth within the panel plugin spec). 

love to hear that this change not only helps UI but also simplifies BE logic 🎉

File: internal/api/shared/migrate/testdata/old_grafana_panels_perses_dashboard.json (hunk 1 above)
Comment (eunicorn) [reply internal/api/shared/migrate/testdata/old_grafana_panels_perses_dashboard.json:51]: I updated my branch to add `omitempty` for `queries`. Can you please pull the new changes?

File: pkg/model/api/v1/dashboard_test.go [hunk 2]
```diff
@@ -285,7 +286,8 @@ func TestMarshalDashboard(t *testing.T) {
                 "up"
               ]
             }
-          }
+          },
+          "queries": null
```
Comment (eunicorn) [inline pkg/model/api/v1/dashboard_test.go:290]: if you pull the new changes where i add `omitempty`, it should undo this change, right?

**Review by eunicorn — APPROVED** [review]
looks good to me! `eunicorn/update-cue-for-panel-spec` is the correct branch!

File: pkg/model/api/v1/dashboard_test.go (hunk 2 above)
Comment (Nexucis) [reply pkg/model/api/v1/dashboard_test.go:290]: yeah totally, this should disappear !

Comment (AntoineThebaud) [reply pkg/model/api/v1/dashboard_test.go]: fixed :)
